### PR TITLE
Ports admin macros and smites from TG

### PR DESCRIPTION
**Pull request by aidan-bowman**
## About the Pull Request

Ports admin macros (for code cleanliness) and admin smiting (for fun).

## Why It's Good For The Game

Generally, the macros TG uses lets them standardize some of the fluff admin calls we have, as well as being easier for newer coders to understand.

Smiting came up as one of the macros, so I thought I would grab it. It lets admins do wacky things to people more easily. I'll add more effects over time.

## Changelog

:cl:
code: Ported admin-related macros from TG.
admin: Ported admin smiting from TG. Turn people into statues with ease!
/:cl:

### CODE REVIEW

**Review by TheDarkElites — CHANGES_REQUESTED** [review]
can you also go through and implement the macros in the code where needed? Thank you :)

**Comment by aidan-bowman** [comment]
> can you also go through and implement the macros in the code where needed? Thank you :)

done

**Review by TheDarkElites — DISMISSED** [review]
Looks fine to me, will also ask Tsurpta to glance over it.

File: code/modules/admin/topic.dm [hunk 1]
```diff
@@ -1182,73 +1182,46 @@
...
 			return
 
-		BSACooldown = 1
-		spawn(50)
-			BSACooldown = 0
-
-		to_chat(M, "You've been hit by bluespace artillery!")
-		log_admin("[key_name(M)] has been hit by Bluespace Artillery fired by [src.owner]")
-		message_staff("[key_name(M)] has been hit by Bluespace Artillery fired by [src.owner]")
+		usr.client.holder.marked_datum_weak = weakref(datum_to_tag)
```
Comment (Tsurupeta) [inline code/modules/admin/topic.dm]: rename it from weak to ref........

Comment (aidan-bowman) [reply code/modules/admin/topic.dm]: Done

**Review by TheDarkElites — DISMISSED** [review]
It looks fine to me, and this has seemingly fixed Tsurpta's concern. Ill ask him for sure before we merge it but for tm this is fine.

**Review by TheDarkElites — APPROVED** [review]
last review is still good for this

**Review by Tsurupeta — APPROVED** [review]

**Pull request merged**